### DISCUSSION TITLE
Conflicting generated source file name

### DISCUSSION THREAD

**Discussion by AndreuCodina** [dotnet/roslyn#69039] [Q&A] [answered]
What source file name can I use? Currently I use

> $"{namespace}.{className}.g.cs"

But for example, I have the folder _Services_ with two files:

**Service1.cs**

```
public partial class Service1<T>
{
  [Exception]
  partial class MyException : Exception;
}
```

**Service2.cs**

```
public partial class Service2<T>
{
  [Exception]
  partial class MyException : Exception;
}
```

Each file has a subclass called MyException, and the generated file name is the same for both.

**Comment by CyrusNajmabadi** [comment, marked as answer]
how about Service1.MyException.cs and Service2.MyException.cs?
